### DISCUSSION TITLE
Use Application access for private application hosted in Kubernetes

### DISCUSSION THREAD

**Discussion by satdeveloping** [gravitational/teleport#15833] [Q&A] [answered]
I know we can use Application access for any ld web app, but what if I don't want to expose my webapp to the public internet.
For example, can I use Application access to expose an ArgoCD deployment in Kubernetes to my Teleport users?

**Comment by webvictim** [comment, marked as answer]
Yes, you'd just need to run a pod inside the cluster which has network access to both the Teleport proxy and the private application endpoint.

The `teleport-kube-agent` Helm chart is a good way to deploy such a pod: https://github.com/gravitational/teleport/tree/master/examples/chart/teleport-kube-agent#manual-configuration-mode

There are also docs for configuring Teleport directly if you install it yourself: https://goteleport.com/docs/application-access/guides/connecting-apps/#start-application-service-with-a-config-file

**Reply by satdeveloping** [reply]
Perfect, thank you
